### PR TITLE
[@types/ember] Add url property to deprecation options

### PR DESCRIPTION
**Pull request by simonihmig**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/emberjs/ember.js/blob/56d60eac537933e20ef66e735299b81dba9a1788/packages/%40ember/debug/lib/deprecate.ts#L16
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by makepanic** [comment]
I think the `url` type is also useful for the ember__application module, see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ember__application/deprecations.d.ts

Can you update it too?

**Review by mike-north — APPROVED** [review]

**Comment by simonihmig** [comment]
> Can you update it too?

Yes, will do so. Don't merge yet!

**Comment by simonihmig** [comment]
Updated `@ember/application/deprecations` as well, ready for another review!

File: types/ember__application/test/deprecations.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@ import { deprecate, deprecateFunc } from '@ember/application/deprecations';
 
 deprecate('this is no longer advised', false, {
     id: 'no-longer-advised',
-    until: 'v4.0'
+    until: 'v4.0',
+    url: 'https://emberjs.com'
```
Comment (makepanic) [inline types/ember__application/test/deprecations.ts]: I'm not sure if it's better, but if you split this call into two where one contains the url and the other doesn't, it keeps the old test while testing the new (optional) behavior.

**Review by mike-north — APPROVED** [review]

**Comment by armanio123** [comment]
Let me know when you feel confident for merging and I can take care of that.

File: types/ember__application/test/deprecations.ts (hunk 1 above)
Comment (simonihmig) [reply types/ember__application/test/deprecations.ts]: Sure, updated it!

**Comment by simonihmig** [comment]
> Let me know when you feel confident for merging and I can take care of that.

From my point of view, this should be ready.

**Comment by simonihmig** [comment]
The failing test seems unrelated. Also fails in master, as well as in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/32770

**Review by makepanic — APPROVED** [review]

**Comment by simonihmig** [comment]
Rebased this again, in the hope that this fixes the (unrelated) test failures.

**Comment by mike-north** [comment]
Tests are currently failing on master w/ TS 3.4 nightlies due to https://github.com/Microsoft/TypeScript/issues/29765 (fix PR: https://github.com/Microsoft/TypeScript/pull/29787)

**Comment by mike-north** [comment]
looks like the TS upstream fix is in. Once a new `dev` release is published, we should see tests passing once again

**Pull request closed without merging**